t/plugin/error-log-skywalking-logger.t

```diff
@@ -0,0 +1,357 @@
+                ngx.HTTP_PUT,
+                [[{
+                    "port": 1999,
+                    "inactive_timeout": 1
+                }]]
+                )
+
+            -- ensure the request is rejected even this plugin doesn't
+            -- have check_schema method
+            ngx.status = code
+            core.log.warn("this is a warning message for test.")
+        }
+    }
+--- request
+GET /tg
+--- error_code: 201
+--- response_body
+--- error_log eval
+qr/please set the correct plugin_metadata for error-log-logger/
+--- wait: 2
+
+
+
+=== TEST 6: test unreachable server
+--- yaml_config
+apisix:
+    enable_admin: true
+    admin_key: null
+plugins:
+  - error-log-logger
+--- config
+    location /tg {
+        content_by_lua_block {
+            local core = require("apisix.core")
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/plugin_metadata/error-log-logger',
+                ngx.HTTP_PUT,
+                [[{
+                    "type": "SKYWALKING",
+                    "skywalking": {
+                        "endpoint_addr": "http://127.0.0.1:1988/log"
+                    },
+                    "inactive_timeout": 1
+                }]]
+                )
+            ngx.sleep(2)
+            core.log.warn("this is a warning message for test.")
+        }
+    }
+--- request
+GET /tg
+--- response_body
+--- error_log eval
+qr/.*\[lua\] batch-processor.lua:63: Batch Processor\[error-log-logger\] failed to process entries: error while sending data to skywalking\[http:\/\/127.0.0.1:1988\/log\] connection refused, context: ngx.timer/
+--- wait: 3
+
+
+
+=== TEST 7: put plugin metadata
+--- yaml_config
+apisix:
+    enable_admin: true
+    admin_key: null
+plugins:
+  - error-log-logger
+--- config
+    location /tg {
+        content_by_lua_block {
+            local core = require("apisix.core")
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/plugin_metadata/error-log-logger',
+                ngx.HTTP_PUT,
+                [[{
+                    "type": "SKYWALKING",
+                    "skywalking": {
+                        "endpoint_addr": "http://127.0.0.1:1982/log"
+                    },
+                    "inactive_timeout": 1
+                }]]
+                )
+        }
+    }
+--- request
+GET /tg
+--- response_body
+--- no_error_log
+[error]
+
+
+
+=== TEST 8: log an error level message
+--- yaml_config
+plugins:
+  - error-log-logger
+--- config
+    location /tg {
+        content_by_lua_block {
+            local core = require("apisix.core")
+            ngx.sleep(2)
+            core.log.error("this is an error message for test.")
+        }
+    }
+--- request
+GET /tg
+--- response_body
+--- error_log eval
+qr/.*\[\{\"body\":\{\"text\":\{\"text\":\".*\"\}\},\"endpoint\":\"\",\"service\":\"APISIX\",\"serviceInstance\":\"APISIX Service Instance\".*/
```

Comment:
Need to check the error log in the text.

Reply:
Also, need a case that multiple error logs are captured and sent.

Reply:
Fixed.